src/text/non_blocking/raw_text_reader.rs

```diff
@@ -37,8 +46,13 @@ pub struct RawTextReader<A: AsRef<[u8]>> {
     // Otherwise, it is None.
     current_ivm: Option<(u8, u8)>,
     bytes_read: usize,
+    // True if the current text buffer is exhausted.
     is_eof: bool,
```

Comment:
I realize this isn't part of your PR, but I wonder if this `is_eof` needs to be tracked separately from just checking `self.buffer.is_empty()` anywhere we currently read it, perhaps via a helper method with a friendly name.